src/learn/concepts/storage.md

```diff
@@ -1 +1,199 @@
+
+## How It Works?
+
+To understand how this model works, let's examine an example where Alice wants to send some amount of private tokens to Bob. Initially, Alice has a record that contains 50 tokens.
+
+![Simple Tx](https://github.com/Aleo-DevRel-Ambas/zero-to-zk/assets/76889160/d3835e13-2cd8-4a67-8e43-b8ba3359b87d)
+
+The record containing 50 tokens is consumed by the program itself, while 2 new records are created by the transition: one to return the unspent amount back to Alice, and another to transfer 30 tokens to Bob.
+
+Remember that Alice has a record containing 20 tokens. Suddenly, she decides to send 5 more tokens to Bob, which will consume the record she previously had and create two new records: one is assigned to Bob, containing 5 tokens, and the other is for Alice, with 15 tokens.
+
+These records will not appear directly in Alice's or Bob's wallets; instead, their clustering point is the Token Program with which they interacted. The final records can be visualized below, using the program as a reference.
+
+![SimpleTxProgramRecord](https://github.com/Aleo-DevRel-Ambas/zero-to-zk/assets/76889160/0ac1d50a-f707-43d7-9483-ba49d67165fa)
+
+This example can extends to more complex programs by creating multiple transitions. Each transaction can include 32 transitions, which all of them are responsible from consuming and creating their own records.
+
+Each account record contains information that specifies the record owner, its stored value, and its application state. Records in Aleo are consumed and newly created from a transition function. A transaction will store multiple transitions, each of which is responsible for the consumption and creation of its individual records. Optionally, if the visibility of the record is private, it can be encrypted using the owner's address secret key.
+
+Transitions can not consume a record that is created by other program. We will examine this later in this section.
+
+## Components of a Record
+
+An Aleo record is serialized in the following format:
+
+| Parameter  |             Type             | Description                                                                                         |
+|------------|:----------------------------:|-----------------------------------------------------------------------------------------------------|
+|     owner    |            address           |                      The address public key of the owner of the program record                      |
+|    data    |    `modifiable`   | A data payload containing arbitrary application-dependent information                               |
+|    nonce   |             group            |                            The serial number nonce of the program record                            |
+| visibility |             enum             | The record's visibility, which can either be public or private. Private if otherwise is not stated. |
```

Comment:
`visibility` can be specify in record struct while writing program but it has no effect to record because record has been default to `private` mode during creation.
Reference: https://github.com/AleoHQ/snarkVM/blob/d48f6fb7d7ca36d134407baebcc59952fe58c546/circuit/program/src/data/record/mod.rs#L55

Reply:
Maybe I referred to wrong code but I did a test using local devnet using record specifying public fields and they are still ciphertext in transaction log.

Reply:
Let's get this moving first and make a future PR for this discussion again if anyone has more opinions on this.